src/tracing/esp32_trace/esp32_tracing.cpp

```diff
@@ -27,6 +27,119 @@ namespace chip {
 namespace Tracing {
 namespace Insights {
 
+namespace {
+static constexpr size_t kPermitListSize = 10;
+static bool gIsPermitListInitialized    = false;
+static size_t gPermitListHashes[kPermitListSize];
+static const char * gPermitListStrings[kPermitListSize] = { "PASESession", "CASESession", "NetworkCommissioning",
```

Comment:
hash and strings seem to be tightly correlated. Why not have a struct of size_t hash, const char *label and then you know you use them at the same time and they are related?

Reply:
Can you please review it once to check if the things are resolved.